### PR TITLE
Update readme to reflect changes to classes

### LINKED ISSUE #335: Update readme for Classes usage
**Enhancement** 

The readme needs an overhaul since the move to classes has been merged (needs to be done before a release).

### PR DESCRIPTION
**Pull request by agubler**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [ ] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [ ] Unit or Functional tests are included in the PR

**Description:**

Updated README to reflect the change to classes

Resolves #335

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,13 +62,16 @@ Constructing your own widgets (Custom widgets) is simple and straightforward.
...
+class MyWidget extends WidgetBase<WidgetProperties> {
+	render() {
        return v('h1', { innerHTML: 'Hello, Dojo!' });
-   }
-});
+	}
+}
 
-createMyWidget.mixin(createProjectorMixin)().append();
+const Projector = ProjectorMixin(MyWidget);
```
Comment (dylans) [inline README.md]: It feels like maybe we should show the import statements for example, but perhaps not

File: README.md [hunk 2]
```diff
@@ -532,35 +533,25 @@ Each widget can have its own locale by passing a property - `properties.locale`.
...
-			}
-		}
-	});
-
-const widget = createI18nWidget({
-	// Set the locale for the widget and all of its children. Any child can
-	// still set its own locale.
-	locale: 'fr'
-});
+class I18nWidget extends I18nMixin(WidgetBase)<I18nWidgetProperties> {
```
Comment (dylans) [inline README.md]: Is it I18n (as state above) or I18nMixin (as stated here)?

File: README.md (hunk 1 above)
Comment (agubler) [inline README.md]: I can include them, there are some included in examples below but not everywhere.

File: README.md [hunk 3]
```diff
@@ -175,13 +174,13 @@ The widget registry allows you to lazy instantiate widgets.
...
-The `createWidgetBase` class provides the functionality needed to create Custom Widgets.
+The `WidgetBase` class provides the functionality needed to create Custom Widgets.
 This functionality includes caching and widget lifecycle management.
 
-The `createWidgetBase` class is available from the `@dojo/widget-core/createWidgetBase` package.
+The `WidgetBase` class is available from the `@dojo/widget-core/WidgetBase ` package.
 
 ```ts
-import { createWidgetBase } from '@dojo/widget-core/createWidgetBase';
+import { WidgetBase } from '@dojo/widget-core/WidgetBase';
```
Comment (rorticus) [inline README.md]: Should this be the default export.. since that will likely be merged soon?

File: README.md (hunk 1 above)
Comment (agubler) [inline README.md]: And yeah the idea is to not keep cluttering examples with imports, but as this is the first one perhaps we should.

File: README.md (hunk 2 above)
Comment (agubler) [inline README.md]: The module is called `I18n` but the exported mixin is called `I18nMixin`.

File: README.md [hunk 4]
```diff
@@ -81,8 +81,7 @@ This code renders a `h1` element onto the page, that says "Hello, Dojo!".
 All widgets in Dojo 2 are designed using key reactive architecture concepts.
 These concepts include unidirectional data flow, inversion of control and property passing.
 
-Dojo 2's widget-core is built using the [@dojo/compose](https://github.com/dojo/compose) composition library.
-This library provides the ability to construct and manipulate traits and mixins.
+Dojo 2's widget-core is built with Typescript leveraging Class mixins to construct and manipulate traits and mixins.
```
Comment (rorticus) [inline README.md]: I think this needs a `,` after `Typescript` ?

File: README.md (hunk 3 above)
Comment (agubler) [inline README.md]: It is named and default. So both are correct, I guess we need to decide which we use consistently through out the documentation.

File: README.md (hunk 4 above)
Comment (agubler) [inline README.md]: I think you are correct.

**Pull request merged**